juju/model.py

```diff
@@ -1774,6 +1774,9 @@ def __init__(self, model):
         self.plan = await self.client_facade.GetBundleChanges(
             yaml.dump(self.bundle))
 
+        if self.plan.errors:
+            raise JujuError(self.plan.errors)
+
     async def execute_plan(self):
```

Comment:
I don't see any check for `self.plan.errors` here in `execute_plan`.  For now, are you confident that this is not an issue because `deploy` will always fetch (and error) before we execute?  Or would it be better to check for plan errors during both fetch and execute?

Reply:
Yes, the only way that the plan is set is via `fetch_plan`, so there's no need to additionally check for errors on execute.